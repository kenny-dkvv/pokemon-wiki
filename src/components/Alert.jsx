/* @jsxImportSource @emotion/react */
import { faTimes } from "@fortawesome/free-solid-svg-icons"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { alert } from "../style"

export const Alert = (props) =>{
  return <div css={alert} style={{background:props.background}}>
    <div>{props.message}</div>
    <FontAwesomeIcon icon={faTimes} onClick={props.closeAlert}/>
  </div>
}