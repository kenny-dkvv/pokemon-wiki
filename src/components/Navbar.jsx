/* @jsxImportSource @emotion/react */
import { navbar, navbarLink } from '../style'
import iconCatchPokemon from '../assets/pokeball_closed.png'
import starterPokemon from '../assets/starter_pokemon.png'
import {Link } from 'react-router-dom';

export const Navbar = () => {
  return <div
  css={navbar}>
    <Link css = {navbarLink} to="/">
      <img src={iconCatchPokemon}/>
      <div>Catch Pokemon</div>
    </Link>
    <Link css = {navbarLink} to="/my-pokemon">
      <img src={starterPokemon} />
      My Pokemon
    </Link>
  </div>
}