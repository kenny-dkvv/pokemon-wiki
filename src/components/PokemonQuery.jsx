/* @jsxImportSource @emotion/react */
import { useQuery } from '@apollo/client';
import pokeballImg from '../assets/pokeball_closed.png'
import { Fragment } from 'react';
import { loadingPokeball, PokemonFragment } from '../style';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTabletAlt } from '@fortawesome/free-solid-svg-icons';
import { GET_POKEMONS } from '../gql_queries';


export const PokemonQuery = (props) => {
  //graphQL fetch
  const { loading, error, data} = useQuery(GET_POKEMONS, {
    variables: props.gqlVariables,
  });

  if (loading) return <div style={{ width: "100%" }}>
    <img loading="lazy" alt={"pokeball"} css={loadingPokeball} src={pokeballImg} />
  </div>

  if (error) return `Error! ${error.message}`

  const myPokemonList = JSON.parse(localStorage.getItem("myPokemon") || "[]");
  
  return <Fragment>
    {
      data.pokemons.results.map((pokemon, idx) => {
        return <Link key={props.gqlVariables.offset + idx + 1} css={PokemonFragment} to={`/pokemon/${pokemon.name}`}>
          <div className="hover-div">
            <img loading="lazy" alt="pokeball" src={pokeballImg} />
            <b>Catch</b>
          </div>

          <img loading="lazy" alt={pokemon.name} src={pokemon.image} />
          <div>{`#${("" + (props.gqlVariables.offset + idx + 1)).padStart(3, "0")}`}</div>
          <div data-testid={`pokemon-${pokemon.name}`}>{`${pokemon.name[0].toUpperCase() + pokemon.name.slice(1)}`}</div>

          <div className="owned">
            <FontAwesomeIcon icon={faTabletAlt} />
            <b>{myPokemonList.filter(pokemon => pokemon.id === props.gqlVariables.offset+idx+1).length }</b>
          </div>

        </Link>
      })
    }
  </Fragment>
}