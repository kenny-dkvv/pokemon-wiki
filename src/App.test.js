// Link.react.test.js
import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { PokemonQuery } from './components/PokemonQuery';
import { render, waitFor } from '@testing-library/react';
import { GET_POKEMONS } from './gql_queries';
import { BrowserRouter } from 'react-router-dom';
const mocks = [
  {
    request: {
      query: GET_POKEMONS,
      variables: {
        offset: 0,
        limit: 20
      },
    },
    result: {
      data: {
        pokemons: {
          results: [
            {
              "url": "https://pokeapi.co/api/v2/pokemon/1/",
              "name": "bulbasaur",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/2/",
              "name": "ivysaur",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/2.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/3/",
              "name": "venusaur",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/4/",
              "name": "charmander",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/5/",
              "name": "charmeleon",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/5.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/6/",
              "name": "charizard",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/6.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/7/",
              "name": "squirtle",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/7.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/8/",
              "name": "wartortle",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/8.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/9/",
              "name": "blastoise",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/9.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/10/",
              "name": "caterpie",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/10.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/11/",
              "name": "metapod",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/11.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/12/",
              "name": "butterfree",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/12.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/13/",
              "name": "weedle",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/13.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/14/",
              "name": "kakuna",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/14.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/15/",
              "name": "beedrill",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/15.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/16/",
              "name": "pidgey",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/16.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/17/",
              "name": "pidgeotto",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/17.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/18/",
              "name": "pidgeot",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/18.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/19/",
              "name": "rattata",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/19.png"
            },
            {
              "url": "https://pokeapi.co/api/v2/pokemon/20/",
              "name": "raticate",
              "image": "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/20.png"
            }
          ]
        },
      },
    },
  },
]

describe("homepage test", () => {

  it('loaded pokemon', async () => {

    let root = render(
      <BrowserRouter>
        <MockedProvider mocks={mocks} addTypename={false} >
          <PokemonQuery gqlVariables={{
            offset: 0,
            limit: 20
          }} />
        </MockedProvider>
      </BrowserRouter>

    );

    
    await waitFor(() => {
      for(let pokemon of mocks[0].result.data.pokemons.results){
        let pokemonNameDOMObject = root.getByTestId(`pokemon-${pokemon.name}`)
        expect(pokemonNameDOMObject.innerHTML).toBe(`${pokemon.name[0].toUpperCase() + pokemon.name.slice(1)}`)
      }
    })
  })

})



