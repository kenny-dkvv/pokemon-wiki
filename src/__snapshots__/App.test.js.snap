// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loaded pokemon 1`] = `
<div
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    data-testid="pokemon-name"
  >
    X
  </div>
  <img
    alt="pokeball"
    className="css-1khj7ka-PokemonQuery"
    loading="lazy"
    src="pokeball_closed.png"
  />
</div>
`;
